wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationRequestStrategy.swift

```diff
@@ -496,23 +496,23 @@ class ConversationByIDTranscoder: IdentifierObjectSyncTranscoder {
     }
 
     func didReceive(response: ZMTransportResponse, for identifiers: Set<UUID>, completionHandler: @escaping () -> Void) {
-        defer { completionHandler() }
 
         guard response.result != .permanentError else {
             if response.httpStatus == 404 {
-                Task {
+                WaitingGroupTask(context: context) { [self] in
                     await deleteConversations(identifiers)
+                    await context.perform { completionHandler() }
```

Comment:
It looks wrong to me, that `completionHandler()` is called inside the `context.perform`. The handler should can manage itself, if core data operations are needed, or not?

Reply:
I think it's to keep the expectation that is called on the syncContext. As the implementation of this method is in multiple places changing the behaviour could involve more work for now.

Reply:
Yeah, it's a bit weird but the handler currently expects the block to be called on the context. I can move the the perform into the handler, it would unnecessary in some cases but this isn't really a performance critical code path and it's easy to forget to to put the `completionHandler()` inside a perform.

Reply:
I moved it the handler then we also only do this in one place.

Reply:
happy we could revert this! 🙌